fix(issues): More adjustments for the page filter bar styling

0Calories opened the pull request:
The page filter bar styling broke again, this PR adds some more CSS rules to further refine the styling. Slightly improves the mobile experience as well

### Before
![image](https://user-images.githubusercontent.com/16740047/229565147-d7378836-6d3f-485e-9b4b-f05813408f7b.png)

(Mobile)
![image](https://user-images.githubusercontent.com/16740047/229565329-97ec9a74-ac66-457f-b98c-d47e9ed275a9.png)


### After
![image](https://user-images.githubusercontent.com/16740047/229565189-cdb68318-7779-49a1-90c3-262701288f60.png)

(Mobile)
![image](https://user-images.githubusercontent.com/16740047/229565428-d58d5e30-64b7-480a-9a84-aadb52d2389d.png)

0Calories committed:
Adjust page filter bar styling

0Calories committed:
Add overflow to label for IssueCategoryFilter

0Calories committed:
Keep date visible at all widths and further adjust

0Calories committed:
Flex adjustments

vuluongj20 commented:
Nice work!

Noticed this small issue while testing:
<img width="703" alt="image" src="https://user-images.githubusercontent.com/44172267/229592387-1ed2cb57-94d0-4e36-b61c-159387766101.png">

0Calories commented on static/app/components/organizations/pageFilterBar.tsx:
```diff
@@ -52,7 +52,7 @@ const PageFilterBar = styled('div')<{condensed?: boolean}>`
 
     /* Prevent date filter from shrinking below 6.5rem */
-    &:last-child {
+    &:nth-last-child(2) {
       min-width: 6.5rem;
     }
   }
```
Ah right, I'll adjust so it's consistent

0Calories committed:
Adjust last-child rule

0Calories commented:
@vuluongj20 I've addressed your comment regarding the `&:last-child` rule, but I'm struggling to find a fix for the dropdowns being stretched out at certain screen widths. It seems to happen only when the screen width goes below `800px` but I can't pinpoint which rule is causing it.

vuluongj20 commented on static/app/views/issueList/filters.tsx:
```diff
@@ -42,12 +42,21 @@ const SearchContainer = styled('div')`
   width: 100%;
-  max-width: 45rem;
-
-  > div > button {
-    width: 100%;
```
@0Calories Looks like this deletion caused the flex issue. Restoring this block fixes the issue.
<img width="703" alt="image" src="https://user-images.githubusercontent.com/44172267/229592387-1ed2cb57-94d0-4e36-b61c-159387766101.png">

0Calories replied:
Ah gotcha, I thought your concern was with the dropdowns being stretched out rather than their normal width. Yeah, restoring that CSS rule fixes the width of the issue category dropdown, thanks 👍

0Calories committed:
Fix width of button
